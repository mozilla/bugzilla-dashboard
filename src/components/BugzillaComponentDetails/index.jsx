import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import DetailView from '../DetailView';

const styles = ({
  subtitle: {
    margin: '0 0 0.5rem',
    color: 'gray',
  },
  metric: {
    display: 'grid',
    gridTemplateColumns: '0.5fr 0.5fr',
  },
  metricLabel: {
    textTransform: 'capitalize',
  },
  metricLink: {
    textAlign: 'center',
  },
});

const BugzillaComponentDetails = ({
  classes, bugzillaEmail, product, component, metrics = {}, onGoBack,
}) => (
  <DetailView title={`${product}::${component}`} onGoBack={onGoBack}>
    <div>
      <h4 className={classes.subtitle}>{bugzillaEmail}</h4>
      {Object.keys(metrics).map(metric => (
        metrics[metric] && (
          <div key={metric} className={classes.metric}>
            <span className={classes.metricLabel}>{metric}</span>
            <a className={classes.metricLink} href={metrics[metric].link}>
              {metrics[metric].count}
            </a>
          </div>
        )
      ))}
    </div>
  </DetailView>

);

BugzillaComponentDetails.propTypes = {
  classes: PropTypes.shape({}).isRequired,
  bugzillaEmail: PropTypes.string.isRequired,
  product: PropTypes.string.isRequired,
  component: PropTypes.string.isRequired,
  metrics: PropTypes.shape({}),
  onGoBack: PropTypes.func.isRequired,
};

BugzillaComponentDetails.defaultProps = {
  metrics: {},
};

export default withStyles(styles)(BugzillaComponentDetails);
