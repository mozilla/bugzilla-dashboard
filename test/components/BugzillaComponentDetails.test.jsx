import React from 'react';
import renderer from 'react-test-renderer';
import BugzillaComponentDetails from '../../src/components/BugzillaComponentDetails';
import bugzillaComponents from '../mocks/bugzillaComponents';

it('renders the details for a Bugzilla component', () => {
  const tree = renderer
    .create((
      <BugzillaComponentDetails
        {...bugzillaComponents[0]}
        onGoBack={() => null}
      />
    ))
    .toJSON();
  expect(tree).toMatchSnapshot();
});
