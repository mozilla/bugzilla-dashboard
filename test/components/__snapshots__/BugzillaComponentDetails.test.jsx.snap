// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the details for a Bugzilla component 1`] = `
<div
  className="DetailView-root-6"
>
  <div>
    <a
      href="/"
      onClick={[Function]}
      rel="noopener noreferrer"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-8"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
    </a>
  </div>
  <div>
    <h2
      className="DetailView-title-7"
    >
      Core::DOM: IndexedDB
    </h2>
    <div>
      <h4
        className="BugzillaComponentDetails-subtitle-1"
      >
        someone@mozilla.com
      </h4>
      <div
        style={
          Object {
            "lineHeight": "8rem",
            "textAlign": "center",
            "width": "8rem",
          }
        }
      >
        <div
          className="MuiCircularProgress-root-18 MuiCircularProgress-colorPrimary-21 MuiCircularProgress-indeterminate-20"
          role="progressbar"
          style={
            Object {
              "height": 40,
              "width": 40,
            }
          }
        >
          <svg
            className="MuiCircularProgress-svg-23"
            viewBox="22 22 44 44"
          >
            <circle
              className="MuiCircularProgress-circle-24 MuiCircularProgress-circleIndeterminate-26"
              cx={44}
              cy={44}
              fill="none"
              r={20.2}
              strokeWidth={3.6}
              style={Object {}}
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
